model/historical_test_data.go

```diff
@@ -16,25 +26,93 @@ type HistoricalTestData struct {
 	Durations       []float64              `bson:"durations"`
 	AverageDuration float64                `bson:"average_duration"`
 	LastUpdate      time.Time              `bson:"last_update"`
+	ArtifactType    PailType               `bson:"artifact_type"`
 
 	env       cedar.Environment
 	populated bool
 }
 
 // CreateHistoricalTestData is an entry point for creating a new
 // HistoricalTestData.
-func CreateHistoricalTestData(info HistoricalTestDataInfo) (*HistoricalTestData, error) {
+func CreateHistoricalTestData(info HistoricalTestDataInfo, artifactStorageType PailType) (*HistoricalTestData, error) {
 	if err := info.validate(); err != nil {
 		return nil, err
 	}
 
+	info.Date = time.Date(
+		info.Date.Year(),
+		info.Date.Month(),
+		info.Date.Day(),
+		0, 0, 0, 0,
+		time.UTC,
+	)
+
 	return &HistoricalTestData{
-		Info:       info,
-		LastUpdate: time.Now(),
-		populated:  true,
+		Info:         info,
+		LastUpdate:   time.Now(),
+		ArtifactType: artifactStorageType,
+		populated:    true,
 	}, nil
 }
 
+// Setup sets the environment. The environment is required for numerous
+// functions on HistoricalTestData.
+func (d *HistoricalTestData) Setup(e cedar.Environment) { d.env = e }
+
+// IsNil returns if the HistoricalTestData is populated or not.
+func (d *HistoricalTestData) IsNil() bool { return !d.populated }
+
+// Find searches the globally configured pail bucket for the
+// HistoricalTestData. The enviromemt should not be nil.
+func (d *HistoricalTestData) Find(ctx context.Context) error {
+	if d.env == nil {
+		return errors.New("cannot find with a nil environment")
+	}
+
+	conf := &CedarConfig{}
+	conf.Setup(d.env)
+	if err := conf.Find(); err != nil {
+		return errors.Wrap(err, "problem getting application configuration")
+	}
+	bucket, err := d.ArtifactType.Create(
+		ctx,
+		d.env,
+		conf.Bucket.HistoricalTestStatsBucket,
+		"",
+		string(pail.S3PermissionsPrivate),
+		true,
+	)
+	if err != nil {
+		return errors.Wrap(err, "problem creating bucket")
+	}
```

Comment:
can we factor out bucket setup (and maybe cache config somewhere?) into a method?